### DISCUSSION TITLE
Queue and other config files - use default connection from database config not 'mysql' string

### DISCUSSION THREAD

**Discussion by xorock** [laravel/framework#49891] [Ideas]
Hello.

In most projects, I use several database connections in the framework, and the first thing I do after setting up a new project is to change the names of these connections, and also rename DB_CONNECTION to DB_DEFAULT_CONNECTION. This means that I have to go through each configuration file and make changes. This leads to strange situations in several places, best seen in the `config/queue.php` file. The connection for the `jobs` table implicitly uses the default connection, but the `failed_jobs` configuration itself will give the connection name `mysql`. The same will be with the recently added `batching` key. A similar situation occurs, for example, in the configuration for telescope.

Wouldn't it be better to use, in all files, the default connection (`config('database.default')`) obtained from the database.php file instead of hardcoded string? I think that 99% of custom cases create migrations on the default connection anyway. And it would greatly clean up the minimalist configuration files in the upcoming Laravel 11.

**config/database.php**
`'default' => env('DB_CONNECTION', 'mysql'),`

**config/queue.php**
```php
'failed' => [
        'driver' => env('QUEUE_FAILED_DRIVER', 'database-uuids'),
        'database' => env('DB_CONNECTION', config('database.default')),  // config('database.default') instead of 'mysql' string
        'table' => 'failed_jobs',
    ],
```

**config/telescope.php**
```php
'storage' => [
        'database' => [
            'connection' => env('DB_CONNECTION', config('database.default')),
            'chunk' => 1000,
        ],
    ],
```

etc.

**Comment by Sophist-UK** [comment]
I agree strongly with this - that the default database connection should be used everywhere by default.

Hard coding "mysql" as a default is database-ist and discriminatory against Postgres etc. We need to be equal handed with all databases and not play favourites like this. 

I might even go so far as to state that using "mysql" rather than the default connection is a bug rather than a feature.

**Reply by shaedrich** [reply]
We don't even have to assume, this is conscious favorite playing, but simply non-DRY anyway already.
